source/load-balancers/tests/sdam-error-handling.yml

```diff
@@ -0,0 +1,192 @@
+description: state change errors are correctly handled
+
+schemaVersion: '1.3'
+
+runOnRequirements:
+  - loadBalanced: true
+
+_yamlAnchors:
+  observedEvents:
+    - &observedEvents
```

Comment:
In the JSON, I see
```
{
      "client": {
        "id": "singleClient",
        "useMultipleMongoses": false,
        "uriOptions": {
          "appname": "lbSDAMErrorTestClient",
          "retryWrites": false
        },
        "observeEvents": [
          "connectionCreatedEvent",
          "connectionReadyEvent",
          "connectionCheckedOutEvent",
          "connectionCheckOutFailedEvent",
          "connectionCheckedInEvent",
          "connectionClosedEvent",
          "poolClearedEvent"
        ]
      }
    },
```

Is this not correct?

Reply:
Ah, that is indeed correct.

I was looking at the JSON for `_yamlAnchors`:

```json
  "_yamlAnchors": {
    "observedEvents": [
      [
        "connectionCreatedEvent",
        "connectionReadyEvent",
        "connectionCheckedOutEvent",
        "connectionCheckOutFailedEvent",
        "connectionCheckedInEvent",
        "connectionClosedEvent",
        "poolClearedEvent"
      ]
    ]
```

I'm aware that JSON is irrelevant but it looks odd to me.

Is the extra array nesting in the YAML actually necessary to resolve correctly? I would have thought this would work:

```yml
_yamlAnchors:
  observedEvents: &observedEvents
    - connectionCreatedEvent
    - connectionReadyEvent
    - connectionCheckedOutEvent
    - connectionCheckOutFailedEvent
    - connectionCheckedInEvent
    - connectionClosedEvent
    - poolClearedEvent
```

Reply:
That seems to work too. Updated in a new commit.